modular_skyrat/modules/ashwalkers/code/buildings/tendril_cursing.dm

```diff
@@ -0,0 +1,59 @@
+/obj/structure/spawner/lavaland
+	/// whether it has a curse attached to it
+	var/cursed = FALSE
+
+/obj/structure/spawner/lavaland/attackby(obj/item/attacking_item, mob/user, params)
+	if(istype(attacking_item, /obj/item/cursed_dagger))
+		playsound(get_turf(src), 'sound/magic/demon_attack1.ogg', 50, TRUE)
+		cursed = !cursed
+		if(cursed)
+			src.add_atom_colour("#41007e", TEMPORARY_COLOUR_PRIORITY)
+		else
+			src.remove_atom_colour(TEMPORARY_COLOUR_PRIORITY, "#41007e")
+		balloon_alert_to_viewers("a curse has been [cursed ? "placed..." : "lifted..."]")
+		if(isliving(user))
+			var/mob/living/living_user = user
+			living_user.adjustBruteLoss(100)
+		to_chat(user, span_warning("The knife sears your hand!"))
+		return
+	return ..()
+
+/obj/structure/spawner/lavaland/Destroy()
+	if(cursed)
+		for(var/mob/living/carbon/human/selected_human in range(7))
```

Comment:
this affects ashies too

Reply:
intended. The curse does not care who killed the tendril but just anyone near it.